WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -730,30 +746,42 @@ static GSourceFuncs _handlerIntervention =
         }
 
     public:
-        virtual string GetHeaders() const final
+#ifdef WEBKIT_GLIB_API
+        uint32_t Headers(string& headers) const override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t Headers(const string& headers) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t UserAgent(string& ua) const override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t UserAgent(const string& ua) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t Languages(string& langs) const override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t Languages(const string& langs) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t LocalStorageEnabled(bool& enabled) const override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t LocalStorageEnabled(const bool enabled) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t HTTPCookieAcceptPolicy(HTTPCookieAcceptPolicyType& policy) const override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t HTTPCookieAcceptPolicy(const HTTPCookieAcceptPolicyType policy) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t BridgeReply(const string& payload) override { return Core::ERROR_UNAVAILABLE; }
+        uint32_t BridgeEvent(const string& payload) override { return Core::ERROR_UNAVAILABLE; }
+#else
+        uint32_t Headers(string& headers) const override
         {
             _adminLock.Lock();
-            string headers = _headers;
+            headers = _headers;
             _adminLock.Unlock();
-            return headers;
+            return Core::ERROR_NONE;
         }
 
-        virtual void SetHeaders(const string& headers) final
+        uint32_t Headers(const string& headers) override
         {
-            if (_context != nullptr) {
-                using SetHeadersData = std::tuple<WebKitImplementation*, string>;
-                auto* data = new SetHeadersData(this, headers);
+            _adminLock.Lock();
```

Comment:
Please keep the headers setting through the SetHeadersData and remove extra locking _adminLock. This avoids deadlocks that we see when the requester tries to set multiple properties and webkit implementation tries to emit a notificatoin

Reply:
I see no harm in moving setting the member to inside the lambda so no problem (certainly if you mention it prevents deadlocks of course :) ), so I will change this